### PR TITLE
Consolidation

### PR DESCRIPTION
**Pull request by michael-e**
This is a big PR, please take a look before pulling it!

I made it v 2.0.0, because I changed the return value of one function. This change breaks everything that relies on this return value. (Shouldn't be a problem in practice, IMHO.)

### CODE REVIEW

File: extension.driver.php [hunk 1]
```diff
@@ -86,8 +90,7 @@ public function install()
 
     public function uninstall()
     {
-        if(parent::uninstall() == true)
-        {
+        if (parent::uninstall() == true) {
```
Comment (nitriques) [inline extension.driver.php]: Can we remove the `== true`?

Comment (michael-e) [inline extension.driver.php]: I think so, but I am a bad programmer.

Comment (michael-e) [inline extension.driver.php]: Should I add that to the PR?

Comment (nitriques) [inline extension.driver.php]: > , but I am a bad programmer.

No you're not! And php does makes a difference between the 2 approaches, it some cases those checks are needed.. But using this approach should be banned

> Should I add that to the PR?

Yes please!

**Comment by michael-e** [comment]
I added a commit. It there is nothing else, I could tag 2.0.1. :-)

I really hope that Nathan is OK with all these changes...

**Comment by michael-e** [comment]
> Why did you removed the reference parameter ?

What do you mean?

**Comment by nitriques** [comment]
> the $result of Symphony's database update.

I've missed that part, sorry.

I need a way to compile, update the DB and get the hash back (that's why I've created the 1.3.0 version).

I also do not see where the boolean result is used in the field's code.. 

But one way we could do it would be to properly set the value in the $entry's data...

**Comment by nitriques** [comment]
> What do you mean?

The & before $entry [here](https://github.com/michael-e/hashid_field/commit/54b1df2abba2c80ddee911d9414cac9ac54ca582#diff-f6c7e7c18f33e001f4b5a3d2b531ba36R220)

**Comment by michael-e** [comment]
Ah, I removed it because the entry object is not changed inside the function. Or is it?

**Comment by michael-e** [comment]
If you need that, we might manipulate the entry object as well. We could do it like in the reflection field: https://github.com/symphonists/reflectionfield/blob/master/fields/field.reflection.php#L385-L398

```php
$data = array(
	'handle'			=> Lang::createHandle($value),
	'value'				=> $value,
	'value_formatted'	=> $value_formatted
);
// Save:
$result = Symphony::Database()->update(
	$data,
	"tbl_entries_data_{$field_id}",
	"`entry_id` = '{$entry_id}'"
);
$entry->setData($field_id, $data);
```

**Comment by michael-e** [comment]
I can add that if you want! We would have both then, a direct database update and a manipulated entry object.

**Comment by nitriques** [comment]
> We could do it like in the reflection field:

yup this is what I was talking about

> Ah, I removed it because the entry object is not changed inside the function. Or is it?

Ah, you're right then!

**Comment by nitriques** [comment]
> We would have both then, a direct database update and a manipulated entry object.

I guess that's the best of both worlds... Still wonder why you need the boolean tho ;)

> I also do not see where the boolean result is used in the field's code..

**Comment by michael-e** [comment]
Indeed the return value is never used inside the extension. This was the reason why I removed the Hashid — it simply did not look straight to me to return this value. And since we are doing a database update in this function, it seems logical and useful to return `false` if it fails.

That said, do you need this in an external script? Could you live with the above modification ("best of both worlds")?

**Comment by nitriques** [comment]
>  it simply did not look straight to me to return this value

Ah!

> it seems logical and useful to return false if it fails.

It does, indeed.

> That said, do you need this in an external script? Could you live with the above modification ("best of both worlds")?

Yes and yes!

**Comment by michael-e** [comment]
Done.

**Comment by michael-e** [comment]
If this is all, I might add a commit to release 2.0.1. Should I?

**Comment by nitriques** [comment]
> Should I?

No since 2.0.0 never went out ;)

**Comment by nitriques** [comment]
I can't test it at the moment, but will do this weekend

**Comment by michael-e** [comment]
> No since 2.0.0 never went out ;)

But it has been commited and tagged… by me…

I will wait until you have tested it. @nathanhornby will hopefully wait with pulling it as well. (You can simply use my branch `consolidation` for testing.)

**Comment by nitriques** [comment]
> But it has been commited and tagged… by me…

But never released!

You can override tags with the -f options ;)

**Comment by michael-e** [comment]
I am closing this and will send a new PR. (In a seperate branch, I removed any release commits and will add a 2.0.0 release commit when the everything is ready to be pulled.)

@nitriques, please use the branch `consolidation2` from [my repo](https://github.com/michael-e/hashid_field) for testing.

**Comment by nitriques** [comment]
Thanks!

**Pull request closed without merging**